qubesmanager/appmenu_select.py

```diff
@@ -81,25 +93,68 @@ def fill_apps_list(self, template=None):
 
         self.app_list.clear()
 
-        command = ['qvm-appmenus', '--get-available',
-                   '--i-understand-format-is-unstable', '--file-field',
-                   'Comment']
+        command = [
+            "qvm-appmenus",
+            "--get-available",
+            "--i-understand-format-is-unstable",
+            "--file-field",
+            "Comment",
+            "--file-field",
+            "Icon",
+        ]
         if template:
-            command.extend(['--template', template.name])
+            command.extend(["--template", template.name])
         command.append(self.vm.name)
 
+        if not hasattr(self.vm, "template"):
+            # TemplateVMs and StandaloneVMs
+            main_template = self.vm.name
+        elif not hasattr(self.vm.template, "template"):
+            # AppVMs
+            main_template = self.vm.template.name
+        else:
+            # DispVMs
+            main_template = self.vm.template.template.name
+
+        template_icons_path = path.join(
+            path.expanduser("~"),
+            ".local",
+            "share",
+            "qubes-appmenus",
+            f"{main_template}",
+            "apps.tempicons",
+        )
+
         try:
-            available_appmenus = [
-                AppListWidgetItem.from_line(line)
-                for line in subprocess.check_output(
-                    command).decode().splitlines()]
+            available_appmenus = []
+            for line in subprocess.check_output(command).decode().splitlines():
+                ident, name, comment, icon_path = line.split("|", maxsplit=4)
+                app_item = AppListWidgetItem.from_line(
+                    "|".join([ident, name, comment])
+                )
+                icon_path = icon_path.replace(
+                    "%VMDIR%/apps.icons", template_icons_path
+                )
+                if path.exists(icon_path):
+                    icon = QtGui.QIcon(icon_path)
+                    qpixmap = icon.pixmap(QtCore.QSize(512, 512))
+                    qimage = QtGui.QImage(qpixmap)
+                    qimage = tint_qimage(qimage, self.vm.label.color)
+                    qpixmap = QtGui.QPixmap(qimage)
+                    icon = QtGui.QIcon(qpixmap)
+                else:
+                    # for `qubes-start.desktop` & missing icons
+                    icon = QtGui.QIcon.fromTheme(self.vm.icon)
```

Comment:
While this fallback case should remain anyway, I think qubes-start.desktop should also have an icon (probably a work for separate PR).